pkg/vmssrollout/drain.go

```diff
@@ -0,0 +1,154 @@
+package vmssrollout
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	policy "k8s.io/api/policy/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const cordonMaxRetries = 5
+
+func (u *VMSSUpgrader) drain(nodeName string) error {
+	var node *v1.Node
+	var err error
+
+	var cordoned bool
+	for i := 0; i < cordonMaxRetries; i++ {
```

Comment:
What the reason we try make node unschedulable few times? Does it fails sometimes?

Reply:
I guess it can fail on a conflict but I haven't seen anything so far. This part is a straight copy from acs-engine (although I continue in all error cases since we only retry 5 times).